From 37daf6aa95882c9d5bdcf3d38786b0f8999a32bb Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@freescale.com>
Date: Thu, 14 Feb 2013 10:24:03 +0200
Subject: [PATCH 052/987] dpaa_eth: Hotfix for cpu_printk() in preemptible
 context

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

cpu_printk() is called, directly or indirectly, from many contexts, not
all of them preempt-safe. For instance, invoking ethtool on a
non-initialized interface while running PREEMPT_RT with
CONFIG_DEBUG_PREEMPT results in an oops.

Temporarily using raw_smp_processor_id() to address this. In the long
run, will get rid of the custom printing macros altogether.

Change-Id: I6a9db7d19cd84ab5b9613b7abde87e85fe0b73bc
Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@freescale.com>
Signed-off-by: Madalin Bucur <madalin.bucur@freescale.com>
(cherry picked from commit dfae81ab1bb804cc335971592242244262bc7b2e)
Reviewed-on: http://git.am.freescale.net:8181/1050
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth-common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth-common.h b/drivers/net/ethernet/freescale/dpa/dpaa_eth-common.h
index eb9c22a..8ea1a3e 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth-common.h
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth-common.h
@@ -63,7 +63,7 @@
 #endif
 
 #define cpu_printk(level, format, arg...) \
-	pr_##level("cpu%d: " format, smp_processor_id(), ##arg)
+	pr_##level("cpu%d: " format, raw_smp_processor_id(), ##arg)
 
 #define cpu_pr_emerg(format, arg...)	\
 	cpu_printk(emerg, format, ##arg)
-- 
1.9.1

